src/registry.rs

```diff
@@ -222,7 +222,28 @@ impl Registry {
 
 // Default registry for rust-prometheus.
 lazy_static! {
-    static ref DEFAULT_REGISTRY: Registry = Registry::default();
+    static ref DEFAULT_REGISTRY: Registry = {
+        let reg = Some(Registry::default());
+
+        match reg {
+            #[cfg(not(feature = "process"))]
+            Some(reg) => {
+                reg
+            },
+
+            // Register process collector.
+            #[cfg(feature = "process")]
+            Some(reg) => {
+                use process_collector::{get_pid, ProcessCollector};
```

Comment:
I think it is more neat.

After all, conditional compilation will also be involved in the new function `new_process_collector`.